From d25adf467a08eeff4ed52a3541e6bce6e2eec90e Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Thu, 24 Nov 2016 09:11:58 +0800
Subject: [PATCH 0924/1566] usb: dwc3: host: inherit the dma_ops from parent
 node

Based on the updates on arm64 dma-mapping, get_dma_ops will return dummy ops
if the device has no related dts nodes, which might be a children device. So
xhci plat initialization process will halt because dma_set_mask_and_coherent
will return EIO.

This patch is to set dma_ops by arch_setup_dma_ops according to its parent's
dma_ops to avoid this error.

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/usb/dwc3/host.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/dwc3/host.c b/drivers/usb/dwc3/host.c
index f6533c6..fb560ab 100644
--- a/drivers/usb/dwc3/host.c
+++ b/drivers/usb/dwc3/host.c
@@ -72,6 +72,9 @@ int dwc3_host_init(struct dwc3 *dwc)
 		return -ENOMEM;
 	}
 
+	if (dwc->dev->archdata.dma_ops)
+		arch_setup_dma_ops(&xhci->dev, 0, 0, NULL, DEV_DMA_COHERENT);
+
 	dma_set_coherent_mask(&xhci->dev, dwc->dev->coherent_dma_mask);
 
 	xhci->dev.parent	= dwc->dev;
-- 
1.7.5.4

